src/core/public/chrome/ui/header/header_logo.tsx

```diff
@@ -74,11 +73,12 @@ function onClick(
   }
 
   if (forceNavigation) {
-    const toParsed = Url.parse(anchor.href);
-    const fromParsed = Url.parse(document.location.href);
+    const toParsed = new URL(anchor.href);
+    const fromParsed = new URL(document.location.href);
     const sameProto = toParsed.protocol === fromParsed.protocol;
     const sameHost = toParsed.host === fromParsed.host;
-    const samePath = toParsed.path === fromParsed.path;
+    const samePath =
+      toParsed.pathname === fromParsed.pathname && toParsed.search === fromParsed.search;
```

Comment:
Why are we adding the `search` match check to the `samePath` bool? If before we were using a combined path and search and calling it a `path`, that's probably something we should fix, by just defining a separate `sameSearch` const and adding it to the conditional. But it's not clear from the limited context here why the search strings must match.

Reply:
Yes. Add a sameSearch condition is more clear here.  Since url.parse includes pathname and search, so I add this condition to make sure use the same logic here. Could we remove the search string compare?

Reply:
Yeah, I'd remove the search string compare.